File: cli/tests/projects/test-frameworks/node/jest/src/calculator.js
/**
 * Calculator module - demonstrates various math operations
 */

export function add(a, b) {
  return a + b;
}

export function subtract(a, b) {
  return a - b;
}

export function multiply(a, b) {
  return a * b;
}

export function divide(a, b) {
  if (b === 0) {
    throw new Error('Division by zero');
  }
  return a / b;
}

export function power(base, exponent) {
  return Math.pow(base, exponent);
}

export function sqrt(n) {
  if (n < 0) {
    throw new Error('Cannot calculate square root of negative number');
  }
  return Math.sqrt(n);
}

export function modulo(a, b) {
  if (b === 0) {
    throw new Error('Modulo by zero');
  }
  return a % b;
}

export function absolute(n) {
  return Math.abs(n);
}

export function factorial(n) {
  if (n < 0) {
    throw new Error('Cannot calculate factorial of negative number');
  }
  if (n === 0 || n === 1) {
    return 1;
  }
  let result = 1;
  for (let i = 2; i <= n; i++) {
    result *= i;
  }
  return result;
}

export function fibonacci(n) {
  if (n < 0) {
    throw new Error('Cannot calculate fibonacci of negative number');
  }
  if (n <= 1) {
    return n;
  }
  let a = 0, b = 1;
  for (let i = 2; i <= n; i++) {
    [a, b] = [b, a + b];
  }
  return b;
}

export function isPrime(n) {
  if (n < 2) return false;
  if (n === 2) return true;
  if (n % 2 === 0) return false;
  for (let i = 3; i <= Math.sqrt(n); i += 2) {
    if (n % i === 0) return false;
  }
  return true;
}

export function gcd(a, b) {
  a = Math.abs(a);
  b = Math.abs(b);
  while (b !== 0) {
    [a, b] = [b, a % b];
  }
  return a;
}

export function lcm(a, b) {
  return Math.abs(a * b) / gcd(a, b);
}

File: cli/tests/projects/test-frameworks/dotnet/xunit/src/XunitApi/Calculator.cs
namespace XunitApi;

/// <summary>
/// Calculator class demonstrating various mathematical operations.
/// </summary>
public class Calculator
{
    public int Add(int a, int b) => a + b;
    
    public int Subtract(int a, int b) => a - b;
    
    public int Multiply(int a, int b) => a * b;
    
    public double Divide(double a, double b)
    {
        if (b == 0)
            throw new DivideByZeroException("Cannot divide by zero");
        return a / b;
    }
    
    public double Power(double baseNum, double exponent) => Math.Pow(baseNum, exponent);
    
    public double Sqrt(double n)
    {
        if (n < 0)
            throw new ArgumentException("Cannot calculate square root of negative number");
        return Math.Sqrt(n);
    }
    
    public int Modulo(int a, int b)
    {
        if (b == 0)
            throw new DivideByZeroException("Cannot calculate modulo by zero");
        return a % b;
    }
    
    public int Factorial(int n)
    {
        if (n < 0)
            throw new ArgumentException("Cannot calculate factorial of negative number");
        if (n <= 1) return 1;
        
        int result = 1;
        for (int i = 2; i <= n; i++)
            result *= i;
        return result;
    }
    
    public int Fibonacci(int n)
    {
        if (n < 0)
            throw new ArgumentException("Cannot calculate fibonacci of negative number");
        if (n <= 1) return n;
        
        int a = 0, b = 1;
        for (int i = 2; i <= n; i++)
        {
            int temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }
    
    public bool IsPrime(int n)
    {
        if (n < 2) return false;
        if (n == 2) return true;
        if (n % 2 == 0) return false;
        
        for (int i = 3; i <= Math.Sqrt(n); i += 2)
        {
            if (n % i == 0) return false;
        }
        return true;
    }
    
    public int Gcd(int a, int b)
    {
        a = Math.Abs(a);
        b = Math.Abs(b);
        while (b != 0)
        {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
    
    public int Lcm(int a, int b) => Math.Abs(a * b) / Gcd(a, b);
}
